File: Pattern/12.py
#      *
#      * *
#      * * *
#      * * * *
#      * * *
#      * *
#      *

n = int(input("Enter the number: "))

# Upper half of the pattern
for i in range(n):
    for j in range(i):
        print("*", end=" ")
    print()

# Lower half of the pattern
for i in range(n, 0, -1): #n - start index, till 0, -1 is iteration(-1 in every row)
    for j in range(i):
        print("*", end=" ")
    print()

File: Patterns in JS/12.js
// 1
// 0 1
// 1 0 1
// 0 1 0 1
// 1 0 1 0 1

function twelve(n) {
  for (let i = 0; i < n; i++) {
    let row = "";
    for (let j = 0; j < i + 1; j++) {
      if ((i + j) % 2 === 0) {
        row += "1 ";
      } else {
        row += "0 ";
      }
    }
    console.log(row);
  }
}

twelve(5);
